management-server/internal/database/app_cache.go

```diff
@@ -0,0 +1,161 @@
+/*
+ *  Copyright (c) 2022, WSO2 LLC. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package database
+
+import (
+	"errors"
+	apkmgt "github.com/wso2/apk/adapter/pkg/discovery/api/wso2/discovery/apkmgt"
+	"github.com/wso2/apk/management-server/internal/logger"
+	"sync"
+	"time"
+)
+
+type CachedApplication struct {
+	application       apkmgt.Application
+	expireAtTimestamp int64
```

Comment:
make this configurable

Reply:
also extend the timestamp in subscription updates

Reply:
fixed in https://github.com/wso2/apk/pull/172/commits/b62377fab07bb4e1fd2b773c8228bafa04c5bb34